linera-views/src/reentrant_collection_view.rs

```diff
@@ -178,6 +178,42 @@ where
     }
 }
 
+impl<C, W> SharableView<C> for ReentrantByteCollectionView<C, W>
+where
+    C: Context + Send + Sync,
+    ViewError: From<C::Error>,
+    W: SharableView<C> + Send + Sync,
+{
+    fn share_unchecked(&mut self) -> Result<Self, ViewError> {
+        let cloned_updates = self
+            .updates
+            .get_mut()
+            .iter()
+            .map(|(key, value)| {
+                let cloned_value = match value {
+                    Update::Removed => Update::Removed,
+                    Update::Set(view_lock) => {
+                        let mut view = view_lock
+                            .try_write()
+                            .ok_or(ViewError::CannotAcquireCollectionEntry)?;
+
+                        Update::Set(Arc::new(RwLock::new(view.share_unchecked()?)))
```

Comment:
I'm still wondering whether we should deep-clone here or actually share the (arc-mutex-ed) subviews.

Reply:
I think a deep clone makes more sense. Otherwise the clone and the source share staging areas for its entries, which would be different from all other views. We can decide otherwise later, or remove `ReentrantCollectionView` entirely before that :man_shrugging:

Reply:
Actually this is the right thing for the actor-model use case.